src/connecting_stream.rs

```diff
@@ -154,34 +154,46 @@ impl ConnectingStream {
         }
     }
 
-    #[cfg(feature = "tls")]
-    fn new_tls_connection(
-        addr: &Url,
-        socket: SelectOk<ConnectingFuture<TcpStream>>,
-        options: &Options,
-    ) -> Self {
+#[cfg(feature = "tls")]
+fn new_tls_connection(
+    addr: &Url,
+    socket: SelectOk<ConnectingFuture<TcpStream>>,
+    options: &Options,
+) -> Result<Self, Box<dyn std::error::Error>> {
         match addr.host_str().map(|host| host.to_owned()) {
-            None => Self {
+            None => Ok(Self {
                 state: State::tls_host_err(),
-            },
+            }),
             Some(host) => {
                 let mut builder = TlsConnector::builder();
                 builder.danger_accept_invalid_certs(options.skip_verify);
+
+                // Load the client certificate and key.
+                let mut client_cert: Vec<u8> = Vec::new();
+                File::open("tls.crt")?.read_to_end(&mut client_cert)?;
```

Comment:
Can you allow the certs and to be passed as options? Ideally using mTLS would be an option a client can decide to use, as is loading the custom CA.